django/db/models/constraints.py

```diff
@@ -119,3 +133,31 @@ def deconstruct(self):
         if self.condition:
             kwargs['condition'] = self.condition
         return path, args, kwargs
+
+
+DEFERRED = 'DEFERRED'
+IMMEDIATE = 'IMMEDIATE'
+
+
+@contextmanager
+def set_constraints(using=None, savepoint=True, **kwargs):
+    connection = get_connection(using)
+    cursor = connection.cursor()
+
+    deferred = ', '.join([name for name, constraint in kwargs.items() if constraint == DEFERRED])
+    immediate = ', '.join([name for name, constraint in kwargs.items() if constraint == IMMEDIATE])
+
+    cleanup = []
+    if deferred:
+        cursor.execute('SET CONSTRAINTS {} DEFERRED'.format(deferred))
+        cleanup.append('SET CONSTRAINTS {} IMMEDIATE'.format(deferred))
+    if immediate:
+        cursor.execute('SET CONSTRAINTS {} IMMEDIATE'.format(immediate))
+        cleanup.append('SET CONSTRAINTS {} DEFERRED'.format(immediate))
+
+    try:
+        with atomic(using=using, savepoint=savepoint):
+            yield
```

Comment:
any hints on what will yield?

Reply:
I think `None`. The `yield` is an implementation detail - we expect this to be used as a `with` without an `as`:

```python
with set_constraints(unique_rank=DEFERRED, unique_name=IMMEDIATE):
    ...
```